From: Matthias Klose <doko@debian.org>
Date: Sat, 29 Oct 2022 21:23:32 -0400
Subject: pydoc: use the pager command if available

Debian file pagers register the "pager" alternative, so if any pager is
available, /usr/bin/pager will exist, and point to the best pager available.
Forwarded: no
---
 Lib/pydoc.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Lib/pydoc.py b/Lib/pydoc.py
index bd64f23..653c87e 100755
--- a/Lib/pydoc.py
+++ b/Lib/pydoc.py
@@ -1582,6 +1582,8 @@ def getpager():
         return plainpager
     if sys.platform == 'win32':
         return lambda text: tempfilepager(plain(text), 'more <')
+    if hasattr(os, 'system') and os.system('(pager) 2>/dev/null') == 0:
+        return lambda text: pipepager(text, 'pager')
     if hasattr(os, 'system') and os.system('(less) 2>/dev/null') == 0:
         return lambda text: pipepager(text, 'less')
 
